Source/Task/AsyncQueue.cpp

```diff
@@ -296,30 +350,31 @@ class Queue
 
     bool Wait(uint32_t timeout)
     {
-#ifdef _WIN32
-        uint32_t w = WaitForSingleObject(m_event, timeout);
-        return (w == WAIT_OBJECT_0);
-#else
-        ASSERT(false);
-        return true;
-#endif
+        std::unique_lock<std::mutex> lock(m_cs);
+        if (!m_hasItems)
+        {
+            m_event.wait_for(lock, std::chrono::milliseconds(timeout));
```

Comment:
are we worried about wait returning early? Condition variables can be spuriously woken by the os

Reply:
Good catch -- added code to handle this.